src/algos/ITI_additional.jl

```diff
@@ -1,274 +1,210 @@
-
-
-
-function euler_residuals(model, s::AbstractArray, x::Array{Array{Float64,2},1}, dr,
-                         dprocess, parms::AbstractArray; with_jres=false, set_dr=true,
-                         jres=nothing, S_ij=nothing)
-
+function euler_residuals(model, s::ListOfPoints, x::Vector{<:ListOfPoints}, dr, dprocess, parms::SVector; keep_J_S=false, set_dr=true) #, jres=nothing, S_ij=nothing)
+    #
     if set_dr ==true
       set_values!(dr,x)
     end
 
-    # s = model.grid.nodes
-    N_s = size(s,1) # Number of gris points for endo_var
-    n_s = size(s,2) # Number of states
-    n_x = size(x[1],2) # Number of controls
-
-    # P = dprocess.values
-    # Q = dprocess.transitions
+    N_s = length(s) # Number of gris points for endo_var
+    n_s = length(s[1]) # Number of states
+    n_x = length(x[1][1]) # Number of controls
```

Comment:
x represents a list of control for each grid today (indexed by the exogenous shock). Currently the type is `Vector{ListOfPoints{n_x}}` but as we discussed somewhere else, it would be useful to create a special type for this object.
`length(x)` is the number of exogenous discretized values (or 1 if zero). `length(x[1])` is number of control variables.

Reply:
It was confusing to me then that `n_x = length(x[1][1])` instead of `length(x[1])`

Reply:
Sorry, it is my comment which was confusing:
- length(x) is number of exogenous values
- length(x[1]) is number of gridpoints (assumed to be constant for now)
- lenght(x[1][1]) is number of control variables